src/app/material-timepicker/components/timepicker-dial-control/ngx-material-timepicker-dial-control.component.ts

```diff
@@ -63,6 +63,12 @@ export class NgxMaterialTimepickerDialControlComponent implements OnInit {
 
     }
 
+    ngOnChanges() {
+        if (this.timeControl) {
+            this.timeControl.setValue(this.formatTimeForUI(this.time));
```

Comment:
Currently, the component's behavior is to only set the form control's value when pressing OK.

I understand this would change that behavior, potentially affecting underlying controls, which would not benefit from the "Cancel" operation anymore.

Is this a correct assumption?

If so, we'd either need to treat it differently, or also update the cancel operation to clear the value once pressed.

It seems to me keeping the current behavior and affecting only the UI instead of the control would be more appropriate.